### PR TITLE
ETS-536 | add type definitions

### PR DESCRIPTION
**Pull request by sunghwan-park-fw**

### CODE REVIEW

File: server/src/auth.ts [hunk 1]
```diff
@@ -5,19 +5,20 @@ import { Strategy as JwtStrategy, ExtractJwt } from 'passport-jwt';
 import { scryptSync } from 'crypto';
 import { dbClient, collections } from './db';
 import logger from './logger';
+import { User } from './types';
```
Comment (hitankar-freshworks) [inline server/src/auth.ts:8]: I suggest we use types folder and create a file for each type

File: server/src/db/db.ts [hunk 2]
```diff
@@ -1,41 +1,14 @@
...
-    /**
-     * Current Database
-     *
-     * @type {Db|null}
-     * @memberof DB
-     */
-    /** eslint-disable-next-line */
-    this.db = null;
-  }
+  private _connection: MongoClient;
```
Comment (hitankar-freshworks) [inline server/src/db/db.ts:10]: What happened to this class?

File: server/src/db/db.ts [hunk 3]
```diff
@@ -109,7 +82,7 @@ class DBClient {
     // and specify the read preference as secondary preferred following AWS best practices
     // https://docs.aws.amazon.com/documentdb/latest/developerguide/connect-to-replica-set.html
     if (useReplicaSet) {
-      options.readPreference = 'secondaryPreferred';
+      options.readPreference = ReadPreference.SECONDARY_PREFERRED;
```
Comment (hitankar-freshworks) [inline server/src/db/db.ts:85]: 💯

File: server/src/db/db.ts (hunk 2 above)
Comment (sunghwan-park-fw) [reply server/src/db/db.ts:10]: I deleted unused or meaningless codes.

File: server/src/auth.ts (hunk 1 above)
Comment (sunghwan-park-fw) [reply server/src/auth.ts:8]: Got it. I'll refactor types.

File: server/package.json [hunk 4]
```diff
@@ -45,8 +45,12 @@
     "@types/express": "4.17.13",
     "@types/jest": "26.0.23",
     "@types/jsonwebtoken": "8.5.5",
-    "@types/mongodb": "4.0.7",
+    "@types/mongodb": "3.6.1",
```
Comment (sunghwan-park-fw) [inline server/package.json:48]: downgraded in order to match it to `mongodb@3.6.1`

**Review by hitankar-freshworks — APPROVED** [review]

**Pull request merged**